cms/static/cms/js/modules/cms.plugins.js

```diff
@@ -134,6 +137,22 @@
                     );
                 });
 
+                // prevents single click from messing up the edit call
```

Comment:
it's very tricky, but similar to what i was thinking. what we could also do is figure out if the `cms-plugin` contains a link/button and show some kind of "edit" icon right after it so the link is clickable and editable. in order for it not to change layout we'd have to position it absolutely somehow. but i think it doesn't play that well with touch and a lot of complicated trickery involved. 

or we keep the behaviour Dmytro did, just add workarounds for ctrlKey and shiftKey to pass through
and if there's js involved - we use http://api.jquery.com/event.stopimmediatepropagation/ which stops other event handlers from executing (although we still need usual stopPropagation in case event is delegated)